src/deserialize/parse.rs

```diff
@@ -271,23 +266,52 @@ fn read_number(
             let mut n = n.chars();
             read_symbol(n.next().unwrap_or(' '), &mut n.enumerate())
         }
+        n if n.to_uppercase().ends_with('N') => Ok(Edn::ArbitraryInt(n)),
+        n if n.to_uppercase().ends_with('M') => Ok(Edn::ArbitraryFloat(n)),
         _ => Err(Error::ParseEdn(format!(
-            "{number} could not be parsed at char count {i} with radix {radix}"
+            "{number} could not be parsed with radix {radix}"
         ))),
     }
 }
 
 fn read_char(chars: &mut std::iter::Enumerate<std::str::Chars<'_>>) -> Result<Edn, Error> {
-    let i = chars
+    let element = chars
         .clone()
-        .next()
-        .ok_or_else(|| Error::ParseEdn("Could not identify symbol index".to_string()))?
-        .0;
-    let c = chars.next();
-    c.ok_or(format!("{c:?} could not be parsed at char count {i}"))
-        .map(|c| c.1)
-        .map(Edn::Char)
-        .map_err(Error::ParseEdn)
+        .enumerate()
+        .take_while(|&(i, c)| i <= 200 && !c.1.is_whitespace())
```

Comment:
Theoretically Clojure supports integers with infinite (memory bounded) size right? Maybe we could find a way to improve/increase this now that we are using string?